src/Templating/Helper/BlockHelper.php

```diff
@@ -84,15 +91,41 @@ class BlockHelper extends Helper
      */
     private $stopwatch;
 
-    public function __construct(BlockServiceManagerInterface $blockServiceManager, array $cacheBlocks, BlockRendererInterface $blockRenderer,
-                                BlockContextManagerInterface $blockContextManager, EventDispatcherInterface $eventDispatcher,
-                                CacheManagerInterface $cacheManager = null, HttpCacheHandlerInterface $cacheHandler = null, Stopwatch $stopwatch = null)
-    {
+    /**
+     * @param CacheManagerInterface|CacheItemPoolInterface|null $cacheManagerOrCachePool
+     */
+    public function __construct(
+        BlockServiceManagerInterface $blockServiceManager,
+        array $cacheBlocks,
```

Comment:
We should document the types allowed in this array, if possible.

Reply:
Not related to this PR, but I can fix this

Reply:
Fixed